neo/ByteArrayComparer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Neo
+{
+    /// <summary>
+    /// Byte array comparer, usefull to have a byte array on a set or as key of a dictionary.
+    /// </summary>
+    public class ByteArrayComparer : IEqualityComparer<byte[]>
+    {
+        /// <inheritdoc />
+        public bool Equals(byte[] left, byte[] right)
+        {
+            if (left == null || right == null)
+            {
+                return left == null && right == null;
+            }
+
+            return left.SequenceEqual(right);
+        }
+
+        /// <inheritdoc />
+        public int GetHashCode(byte[] key)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException("key");
+            }
+
+            return key.Sum(b => b);
```

Comment:
We can use murmur instead. This class is general purpose, but we could possibly index everything in this case by uint256, because all MPT keys are 32byte sha256.